packages/web3-wrapper/src/index.ts

```diff
@@ -155,13 +130,17 @@ export class Web3Wrapper {
         const formattedLogs = _.map(rawLogs, this.formatLog.bind(this));
         return formattedLogs;
     }
-    private getContractInstance(abi: Web3.ContractAbi, address: string): Web3.ContractInstance {
-        const web3ContractInstance = this.web3.eth.contract(abi).at(address);
+    public getContractFromAbi(abi: Web3.ContractAbi): Web3.Contract<any> {
+        const web3Contract = this.web3.eth.contract(abi);
+        return web3Contract;
+    }
+    public getContractInstance(abi: Web3.ContractAbi, address: string): Web3.ContractInstance {
+        const web3ContractInstance = this.getContractFromAbi(abi).at(address);
         return web3ContractInstance;
     }
-    private async getNetworkAsync(): Promise<number> {
-        const networkId = await promisify<number>(this.web3.version.getNetwork)();
-        return networkId;
+    public async estimateGasAsync(data: string): Promise<number> {
+        const gas = await promisify<number>(this.web3.eth.estimateGas)({data});
+        return gas;
```

Comment:
We should re-wrap `gas` in a new BigNumber.js before returning.

Reply:
Rewrap a number and make it a better number?

Reply:
ah nvm